files/static/index.js

```diff
@@ -0,0 +1,340 @@
+          !domain.includes("..");
+  },
+
+  // hasLikelyETLD1: function(domain) {
+  //   // Submitting .com domains is common; don't use the fast path for .co
+  //   // Same for .net => .ne
+  //   if (domain.endsWith(".co") || domain.endsWith(".ne")) {
+  //     return true;
+  //   }
+
+  //   var publicsuffix = tldjs.getPublicSuffix(domain)
+  //   return publicsuffix !== null && publicsuffix !== domain;
+  // },
+
+  clearTheme: function(theme) {
+    document.body.classList.remove("theme-green", "theme-yellow", "theme-red");
+  },
+
+  setTheme: function(theme) {
+    this.clearTheme();
+    document.body.classList.add(theme)
+  },
+
+  currentDomain: function() {
+    var domain = $("#domain").value;
+    // Check for pasted URLs beginning with http:// or https://
+    if (domain.startsWith("http://")) {
+      domain = domain.slice("http://".length)
+    }
+    if (domain.startsWith("https://")) {
+      domain = domain.slice("https://".length)
+    }
+    return domain;
+  },
+
+  clearResults: function() {
+    $("#result").classList.add("hidden");
+  },
+
+  showResults: function(domain, checkDomainIssues, domainStatus) {
+    if (checkDomainIssues.errors.length === 0) {
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#summary").textContent = "" + domain + " satisfies all requirements for preloading!";
+        this.setTheme("theme-green");
+      } else {
+        $("#summary").textContent = "" + domain + " has " + (checkDomainIssues.warnings.length == 1 ? "a warning" : "warnings") + ", but satisfies all requirements for preloading.";
+        this.setTheme("theme-yellow");
+      }
+    } else {
+        $("#summary").textContent = "" + domain + " has errors and cannot be preloaded.";
+        this.setTheme("theme-red");
+    }
+
+    switch (domainStatus.status) {
+    case "unknown":
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent = "" + domain + " is not yet preloaded, and may be submitted to the preload list."
+      } else {
+        $("#status").textContent = "" + domain + " is not preloaded."
+      }
+      break;
+    case "pending":
+      $("#status").textContent = "" + domain + " is pending submission to the preload list."
+      break;
+    case "preloaded":
+      $("#status").textContent = "" + domain + " is currently preloaded."
+      break;
+    case "rejected":
+      // TODO: do something useful if domainStatus.message is not present.
+      if (domainStatus.message) {
+        $("#status").textContent = "" + domain + " was rejected from the preload list for the following reason: " + domainStatus.message;
+      } else {
+        $("#status").textContent = "" + domain + " was rejected from the preload list.";
+      }
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent += " It may be submitted again."
+      }
+      break;
+    case "removed":
+      if (checkDomainIssues.warnings.length === 0) {
```

Comment:
Any reason not to do if (!checkDomainIssues.warnings) instead? (here and elsewhere)